File: crates/klyron_workspace/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron::_workspace {

WorkspaceBuilder::WorkspaceBuilder() : config_{} {}

WorkspaceBuilder& WorkspaceBuilder::set_config(const WorkspaceConfig& cfg) {
    config_ = cfg;
    return *this;
}

WorkspaceConfig WorkspaceBuilder::config() const {
    return config_;
}

}

File: crates/klyron_workspace/bindings/rust/builder.rs
//! Builder for klyron_workspace
use crate::types::WorkspaceConfig;

pub struct WorkspaceBuilder {
    config: WorkspaceConfig,
}

impl WorkspaceBuilder {
    pub fn new() -> Self {
        Self { config: WorkspaceConfig::default() }
    }
    pub fn with_version(mut self, version: &str) -> Self {
        self.config.version = version.to_string();
        self
    }
    pub fn build(self) -> WorkspaceConfig {
        self.config
    }
}
